cmd/controller-manager/main.go

```diff
@@ -79,12 +81,29 @@ func main() {
 
 	ctrl.SetLogger(zap.Logger(true))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:             scheme,
-		MetricsBindAddress: common.ControllerCfg.MetricsAddr,
-		LeaderElection:     common.ControllerCfg.EnableLeaderElection,
-		Port:               9443,
-	})
+	var mgr manager.Manager
+	var err error
+
+	if common.ControllerCfg.ClusterScoped {
+		setupLog.Info("Chaos controller manager is running in cluster scoped mode.")
+		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+			Scheme:             scheme,
+			MetricsBindAddress: common.ControllerCfg.MetricsAddr,
+			LeaderElection:     common.ControllerCfg.EnableLeaderElection,
+			Port:               9443,
+		})
+	} else {
+		setupLog.Info("Chaos controller manager is running in namespace scoped mode.", "namespace", common.ControllerCfg.Namespace)
+		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+			Scheme: scheme,
+			// one more thing is the specific namespace
+			Namespace:          common.ControllerCfg.Namespace,
+			MetricsBindAddress: common.ControllerCfg.MetricsAddr,
+			LeaderElection:     common.ControllerCfg.EnableLeaderElection,
+			Port:               9443,
+		})
+	}
```

Comment:
```
var ns string = ""
if common.ControllerCfg.ClusterScoped {
		setupLog.Info("Chaos controller manager is running in cluster scoped mode.")
	} else {
		setupLog.Info("Chaos controller manager is running in namespace scoped mode.", "namespace", common.ControllerCfg.Namespace)
                 ns = common.ControllerCfg.Namespace
}

mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
			Scheme: scheme,
			Namespace:          ns,
			MetricsBindAddress: common.ControllerCfg.MetricsAddr,
			LeaderElection:     common.ControllerCfg.EnableLeaderElection,
			Port:               9443,
		})
```

I think it's better that way.  Default watches all namespaces.

Reply:
I agree with U :)